crates/net/eth-wire/src/types/message.rs

```diff
@@ -468,8 +468,26 @@ where
     T: Decodable,
 {
     fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
-        let _header = Header::decode(buf)?;
-        Ok(Self { request_id: u64::decode(buf)?, message: T::decode(buf)? })
+        let header = Header::decode(buf)?;
+        // Check that the buffer is large enough for the payload length
+        if buf.len() < header.payload_length {
```

Comment:
@Rjected I don't think this is necessary, because `Header::decode` check it already. 

https://github.com/alloy-rs/rlp/blob/main/crates/rlp/src/header.rs#L71-L73

Reply:
yeah this looks redundant

Reply:
> @Rjected I don't think this is necessary, because `Header::decode` check it already.
> 
> https://github.com/alloy-rs/rlp/blob/main/crates/rlp/src/header.rs#L71-L73

Right, then let's just do the consumption check

Reply:
Done.